src/app/services/groupsocket/groupsocket.service.ts

```diff
@@ -81,6 +91,12 @@ export class GroupSocketService {
   leaveGroup() {
     this.userInfo.socketID = this.socket.io.engine.id;
     this.socket.emit('leaveGroup', this.userInfo);
+    //Remove all userInfos from array except for the current user's userInfo.
+    for (var i = this.userInfos.length - 1; i >= 0; i--) {
+      if (this.userInfos[i].socketID != this.userInfo.socketID) {
```

Comment:
I think instead of this forLoop here, we could use the `filter` function instead. It would be a lot cleaner and simpler.

Reply:
```
this.userInfos = this.userInfos.filter(userInfo => userInfo.socketID === this.userInfo.socketID);
```
Should remove everything except for the current user's info.

Reply:
Good idea. Surprised I didn't think of that hehe. Thanks!